controllers/feeds/getFeedChat.js

```diff
@@ -152,16 +153,39 @@ const finalize = (req, id, myReaction, newGroup, activity_id, constantActor) =>
 
 const getFeedGroup = async (groupingFeed) => {
   const feedGroup = [];
-  // eslint-disable-next-line no-restricted-syntax
+  const feedIds = [];
   for (const feed of groupingFeed) {
-    // eslint-disable-next-line no-await-in-loop
-    const blockCount = await UserBlockedUser.count({
-      where: {
-        post_id: feed.activity_id
-      }
+    feedIds.push(feed.activity_id);
+  }
+
+  let resultBlockCount = [];
+  try {
+    const queryBlockCount = `
+    SELECT post_id, COUNT(post_id) AS COUNT
+    FROM user_blocked_user
+    WHERE post_id IN (:feedIds)
+    GROUP BY post_id`;
+
+    resultBlockCount = await sequelize.query(queryBlockCount, {
+      replacements: {
+        feedIds
+      },
+      type: Sequelize.QueryTypes.SELECT
     });
-    feedGroup.push({...feed, block: blockCount});
+  } catch (e) {
+    console.error('Failed to get count block users', e);
+    throw e;
+  }
+
+  for (const feed2 of groupingFeed) {
+    const filteredFeed = resultBlockCount.filter(
+      (feedDetail) => feed2.activity_id === feedDetail.post_id
+    );
+    const blockCount = filteredFeed[0] ? parseInt(filteredFeed[0].count) : 0;
+
+    feedGroup.push({...feed2, block: blockCount});
   }
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for getting the block count has been improved by replacing multiple database queries with a single raw SQL query. However, there's a potential SQL injection vulnerability in this code. Although Sequelize escapes replacements by default when you use the `:key` syntax, it's still good practice to validate or sanitize input data before using it in a SQL query. In this case, ensure that `feedIds` only contains valid IDs.

Also, the error handling could be improved. Currently, if the query fails, the error is logged and then re-thrown, but this will stop the execution of the entire function. Depending on your application's requirements, you might want to handle this error more gracefully, for example by returning an empty array or a specific error message.

Lastly, the filtering operation at lines 180-184 can be optimized. Instead of filtering `resultBlockCount` for each feed, consider converting `resultBlockCount` into a map (object) where the keys are the post_ids and the values are the counts. This way, you can look up the block count for each feed in constant time.

Here's how you could implement these suggestions:

```diff
  let resultBlockCount = [];
  try {
    const queryBlockCount = `
    SELECT post_id, COUNT(post_id) AS COUNT
    FROM user_blocked_user
    WHERE post_id IN (:feedIds)
    GROUP BY post_id`;

+   // Ensure feedIds only contains valid IDs
+   const validFeedIds = feedIds.filter(id => typeof id === 'number' || (typeof id === 'string' && /^\d+$/.test(id)));
    resultBlockCount = await sequelize.query(queryBlockCount, {
      replacements: {
-       feedIds
+       feedIds: validFeedIds
      },
      type: Sequelize.QueryTypes.SELECT
     });
  } catch (e) {
    console.error('Failed to get count block users', e);
-   throw e;
+   return [];
  }

+ // Convert resultBlockCount into a map for faster lookups
+ const blockCountMap = {};
+ for (const { post_id, COUNT } of resultBlockCount) {
+   blockCountMap[post_id] = parseInt(COUNT);
+ }

  for (const feed2 of groupingFeed) {
-   const filteredFeed = resultBlockCount.filter(
-     (feedDetail) => feed2.activity_id === feedDetail.post_id
-   );
-   const blockCount = filteredFeed[0] ? parseInt(filteredFeed[0].count) : 0;
+   const blockCount = blockCountMap[feed2.activity_id] || 0;

    feedGroup.push({...feed2, block: blockCount});
  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->